DemoServer/Controllers/Demos/StaticIndexes/AdditionalSourcesIndex/AdditionalSourcesIndexController.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Threading.Tasks;
+using DemoCommon.Models;
+using DemoServer.Utils.Cache;
+using DemoServer.Utils.Database;
+using DemoServer.Utils.UserId;
+using Microsoft.AspNetCore.Mvc;
+#region Usings
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using Raven.Client.Documents.Session;
+using Raven.Client.Documents.Indexes;
+#endregion
+
+namespace DemoServer.Controllers.Demos.StaticIndexes.AdditionalSourcesIndex
+{
+    public class AdditionalSourcesIndexController : DemoCodeController
+    {
+        public AdditionalSourcesIndexController(UserIdContainer userId, UserStoreCache userStoreCache, MediaStoreCache mediaStoreCache,
+            DatabaseSetup databaseSetup) : base(userId, userStoreCache, mediaStoreCache, databaseSetup)
+        {
+        }
+        
+        #region Demo
+        #region Step_1
+        public class Products_ByPrice : AbstractIndexCreationTask<Product, Products_ByPrice.IndexEntry>
+            #endregion
+        {
+            #region Step_2
+            public class IndexEntry
+            {
+                public string ProductName { get; set; }
+                public decimal OriginalPrice { get; set; }
+                public decimal SalePrice { get; set; }
+                public decimal ProfitPrice { get; set; }
+            }
+            #endregion
+
+            public Products_ByPrice()
+            {
+                #region Step_3
+                Map = products => from product in products
+                    select new IndexEntry
+                    {
+                        ProductName = product.Name,
+                        OriginalPrice = product.PricePerUnit,
+                        SalePrice = DiscountUtils.CalcSalePrice(product.PricePerUnit),
+                        ProfitPrice = DiscountUtils.CalcProfitPrice(product.PricePerUnit)
+                    };
+                #endregion
+                
+                #region Step_4
+                AdditionalSources = new Dictionary<string, string>
+                {
+                    {
+                        "DiscountLogic", System.IO.File.ReadAllText(
+                            Path.Combine(AppContext.BaseDirectory, "..", "..", "..", "MyAdditionalSourceFile.cs"))
```

Comment:
would it be possible to show the content of this file somehow? I think it would be very helpful in order to comprehend what is being calculated and how and the level of freedom and possibilities that are an option here

Reply:
done - added code in comments to the file - I think it looks ok
(it didn't look good when added in the walkthrough)

![image](https://user-images.githubusercontent.com/17376974/151436763-b75ef305-98ab-4647-b8f3-9ef042d26931.png)